std/stdio.d

```diff
@@ -307,9 +307,13 @@ public:
     }
 }
 
+/// @@@DEPRECATED_2019-01@@@
+deprecated("Use .byRecord")
+alias ByRecord = ByRecordImpl;
```

Comment:
Stupid question: Does alias make `ByRecord` public even though it's private? I seem to remember even if you public aliased a private symbol, it still didn't work.

Reply:
I just tested this:
````
// mod.d
module mod;
private struct S {}
alias PubS = S;

// test.d
import mod;
void main() {
        PubS s; // OK
        //S t;  // NG
}
````
Compiler accepts the first declaration `s`, but rejects the declaration `t`. So we're good.

Reply:
I thought that that was the case, but some investigation shows that that's not quite right - e.g. [this comment](https://issues.dlang.org/show_bug.cgi?id=4533#c7) by Marten on an old enhancement request indicates that it works except for functions, and testing it, that seems to be the case. I get an error if I try to use a public alias of a private function, but it works with a variable or type.

However, I would point out that the comment shouldn't be a ddoc comment. `ByRecord` isn't documented, and this then documents it. The same goes for the other comments. They should be regular comments, not documentation comments.

Reply:
And H.S.Teoh replied while I was in the middle of my own reply. :)

Reply:
I just tested this as well by simply inserting `ByLine!(char, char)` into a random Phobos module:

```
std/file.d(4384): Deprecation: std.stdio.File.ByLine!(char, char).ByLineImpl is not visible from module file
```

Hence, I used the `alias this` trick ->


```
std/file.d(4384): Deprecation: std.stdio.File.ByLine!(char, char).ByLineImpl is not visible from module file
```

Reply:
Is there supposed to be a difference between those two?

Reply:
ok, I see it now.

Reply:
Copy/pasting mistake - sorry.

Reply:
Hm... maybe it's because it's a template. In any case, the alias this trick works.